alchemist-cognitive-agents/src/test/kotlin/it/unibo/alchemist/test/TestOrientingBehavior.kt

```diff
@@ -11,40 +11,46 @@ package it.unibo.alchemist.test
 
 import io.kotest.core.spec.style.StringSpec
 import io.kotest.matchers.collections.beEmpty
+import io.kotest.matchers.collections.shouldNotBeIn
 import io.kotest.matchers.comparables.shouldBeLessThan
 import io.kotest.matchers.shouldBe
 import io.kotest.matchers.shouldNot
 import io.kotest.matchers.shouldNotBe
 import it.unibo.alchemist.model.interfaces.Environment
 import it.unibo.alchemist.model.interfaces.NavigationAction
 import it.unibo.alchemist.model.interfaces.NavigationStrategy
-import it.unibo.alchemist.model.interfaces.OrientingPedestrian
+import it.unibo.alchemist.model.interfaces.Node
 import it.unibo.alchemist.model.interfaces.Position
+import it.unibo.alchemist.model.interfaces.properties.OrientingProperty
 import it.unibo.alchemist.model.interfaces.environments.Euclidean2DEnvironmentWithGraph
 import it.unibo.alchemist.model.interfaces.geometry.Vector
 import it.unibo.alchemist.testsupport.loadYamlSimulation
 import it.unibo.alchemist.testsupport.startSimulation
 import org.apache.commons.collections4.queue.CircularFifoQueue
+import it.unibo.alchemist.model.interfaces.Node.Companion.asPropertyOrNull
 
 /**
  * Contains tests concerning [NavigationAction]s and [NavigationStrategy], such tests are
  * dependent on the pedestrian's speed.
  */
 class TestOrientingBehavior<T, P> : StringSpec({
 
+    val filterOrientingNode: (Node<T>) -> Boolean =
+        { it.asPropertyOrNull<T, OrientingProperty<T, P, *, *, *, *>>() != null }
```

Comment:
if this is used multiple times, transform it in a function. Possibly, an `Iterable<Node<T>>.() -> Iterable<OrientingProperty<.....>>` function. Otherwise, drop it